test/test_jit_cuda_fuser.py

```diff
@@ -427,7 +427,44 @@ def _compare(self, desc, inp1, inp2, error):
             print("inp2   : ", b[index])
         return close
 
-    def _reduction_helper(self, sizes, reduction_axis, dtype, device):
+    def _permutation_helper(self, sizes, broadcast_axis, dtype, device, perm0, perm1):
+        class MyBinaryOps(torch.nn.Module):
+            def forward(self, x: torch.Tensor, y: torch.Tensor):
+                o = torch.add(x, y)
+                o = torch.relu(o)
+                return o
+
+        t = MyBinaryOps()
+
+        x = torch.randn([sizes[i] for i in perm0], dtype=dtype, device=device).permute([perm0.index(i) for i in range(len(sizes))])
+        sizes[broadcast_axis] = 1
+        y = torch.randn([sizes[i] for i in perm1], dtype=dtype, device=device).permute([perm1.index(i) for i in range(len(sizes))])
+        t_jit = torch.jit.script(t)
+        jit_o = t_jit(x, y)
+        jit_o = t_jit(x, y)
+        o = t(x, y)
+        for oo, jit_oo in zip(o, jit_o):
+            self.assertEqual(oo.dtype, jit_oo.dtype)
+            # numerical issues here due to our scheduling.
+            # can't use `self.assertEqual(oo, jit_oo)`
+            self.assertTrue(self._compare("comparing output failed", oo, jit_oo, 1e-4))
```

Comment:
1e-4? seems large, no?

Reply:
Good catch, that 1e-4 was used for the reduction cases, which was copied here and is totally not needed.
On the note, for reduction we should be able to relax the error check once I switched to Kevin's scheduler.